### ISSUE TITLE
Provide a way to provide custom protobuf marshallers

### ISSUE THREAD

**Issue by jrhee17** [line/armeria#5103]
It has been pointed out that we do not provide a way to override the default marshaller for protobuf application types.

The following marshallers are not used to actually parse/serialize the messages.
https://github.com/line/armeria/blob/ce5b42f36a4a5515a28cee7a15b2a3e2ca5f76c2/grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/GrpcMessageMarshaller.java#L83-L84

This makes it difficult for users to specify their own marshaller.

We may provide an option to override the marshaller used internally, similarly to how we did for `GrpcJsonMarshaller`

ref: https://discord.com/channels/1087271586832318494/1087272728177942629/1137106282243969075

**Comment by ikhoon** [comment]
AFAIK, `Message.writeTo()` is used to directly convert protobuf messages to `ByteBuf` to avoid an additional bytes copy.
https://github.com/line/armeria/blob/b8e54200ed28e14f8a31867f576e9a8942b5da6b/grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/GrpcMessageMarshaller.java#L202-L203
I think we may add an option to respect the marshaller specified in gRPC `MethodDescriptor`.
```java
GrpcService
  .builder()
  // Disabled by default
  // When enabled, respect `MethodDescriptor.Marshaller`.
  .useMethodMarshaller(true)
```

> We may provide an option to override the marshaller used internally, similarly to how we did for GrpcJsonMarshaller

It would a good idea if `GrpcServiceBuilder` provides a shortcut API to set `Marshaller` without `ServerInterceptors`.

**Comment by minwoox** [comment]
Closed by #5630
